src/tuesday/core/psclasses.py

```diff
@@ -0,0 +1,31 @@
+"""Classes to hold the power spectrum data."""
+
+from dataclasses import dataclass
+
+import astropy.units as un
+import numpy as np
+
+
+@dataclass(frozen=True)
+class SphericalPS:
+    """Class to hold the 1D power spectrum data."""
+
+    ps: un.Quantity
+    k: un.Quantity
+    redshift: np.ndarray | None = None
+    Nmodes: np.ndarray | None = None
+    var: un.Quantity | None = None
+    delta: bool = False
```

Comment:
This is good, but a couple of suggestions:

make `Nmodes` instead `n_modes`. Make `var` -> `variance` and `delta` -> `is_deltasq`.

Also, add a `__post_init__` that does some validation (checking array sizes etc).